services/worldstatestore/fake/fake.go

```diff
@@ -311,142 +571,317 @@ func (f *WorldStateStore) initializeStaticTransforms() {
+				}
+			}
+		}
+	}
+	worldState.mu.Unlock()
+
+	updatedFields := []string{
+		"physicalObject.geometryType.value.pointCloud.pointCloud",
+		"physicalObject.geometryType.value.pointCloud.header",
 	}
+
+	deltaTransform := &commonpb.Transform{
+		Uuid: transform.Uuid,
+		PhysicalObject: &commonpb.Geometry{
+			GeometryType: &commonpb.Geometry_Pointcloud{
+				Pointcloud: &commonpb.PointCloud{
+					PointCloud: chunkData,
+					Header:     header,
+				},
+			},
+		},
+	}
+
+	worldState.emitTransformChange(deltaTransform, pb.TransformChangeType_TRANSFORM_CHANGE_TYPE_UPDATED, updatedFields)
 }
 
-func (f *WorldStateStore) emitTransformUpdate(partial *commonpb.Transform, updatedFields []string) {
-	if partial == nil || len(partial.GetUuid()) == 0 {
+func (worldState *WorldStateStore) emitTransformChange(transform *commonpb.Transform, changeType pb.TransformChangeType, updatedFields []string) {
+	if transform == nil || len(transform.GetUuid()) == 0 {
 		return
 	}
+
 	change := worldstatestore.TransformChange{
-		ChangeType:    pb.TransformChangeType_TRANSFORM_CHANGE_TYPE_UPDATED,
-		Transform:     partial,
+		ChangeType:    changeType,
+		Transform:     transform,
 		UpdatedFields: updatedFields,
 	}
+
 	select {
-	case f.changeChan <- change:
-	case <-f.streamCtx.Done():
+	case worldState.changeChan <- change:
+		// Successfully sent
+	case <-worldState.streamCtx.Done():
+		return
 	default:
-		// Channel is full, skip this update
+		// Channel is full - implement backpressure strategy with brief timeout
+		select {
+		case worldState.changeChan <- change:
+		case <-time.After(time.Millisecond):
+			// Drop update after timeout to prevent blocking
+		case <-worldState.streamCtx.Done():
+			return
+		}
 	}
 }
 
-func (f *WorldStateStore) animationLoop() {
-	f.mu.RLock()
-	curFPS := f.fps
-	f.mu.RUnlock()
-	if curFPS <= 0 {
-		curFPS = 1
-	}
-	interval := time.Duration(float64(time.Second) / curFPS)
-	ticker := time.NewTicker(interval)
+func (worldState *WorldStateStore) animate() {
+	var (
+		curFPS          = 10.0
+		interval        = time.Duration(float64(time.Second) / curFPS)
+		ticker          = time.NewTicker(interval)
+		lastElapsed     time.Duration
+		fpsCheckCounter int64
+	)
 	defer ticker.Stop()
 
 	for {
 		select {
-		case <-f.streamCtx.Done():
+		case <-worldState.streamCtx.Done():
 			return
 		case <-ticker.C:
-			f.updateTransforms()
-			// Reconfigure ticker if FPS changed
-			f.mu.RLock()
-			newFPS := f.fps
-			f.mu.RUnlock()
-			if newFPS != curFPS && newFPS > 0 {
-				ticker.Stop()
-				curFPS = newFPS
-				interval = time.Duration(float64(time.Second) / curFPS)
-				ticker = time.NewTicker(interval)
+			elapsed := time.Since(worldState.startTime)
+			if elapsed-lastElapsed < time.Millisecond {
+				continue
+			}
+
+			lastElapsed = elapsed
+			worldState.update()
+
+			fpsCheckCounter++
+			if fpsCheckCounter%100 == 0 {
+				worldState.mu.RLock()
+				newFPS := worldState.fps
```

Comment:
would also probably call anything FPS related "ticksPerSecond" or "stepsPerSecond" since there's no rendering in this context but that's a bigger lift